### DISCUSSION TITLE
Is it possible to resume the blocked system call?

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#185] [Project Q&A 2. Transport Layer (TCP)]
Hello.

I was implementing blocked system calls in connect() and accept().
I understood blocked system calls like the following: 
If we return from a system call without returnSystemCall() method in a certain conditional statement, the application is blocked until the condition becomes false.  

Then how can we explicitly return a blocked system call resume the syscall after escaping from the conditional statement?
Does it require calling returnSystemCall() method?

Thanks in advance.

**Comment by Pusnow** [comment]
When handling system calls, you should store `syscallUUID`. TCPAssignment's member variable would be perfect place for it.

Later, in `packetArrived()` or `timerFired()`, you can check the conditional statement and unblock the system calls using the stored UUID.

**Reply by ytrewq271828** [reply]
Then are there any specific return values of the returnSystemCall() when unblocking the calls?

**Reply by Pusnow** [reply]
They depend on system calls. You should follow RETURN VALUE sections in manuals.

e.g.,

https://github.com/ANLAB-KAIST/KENSv3/blob/eba97b6c81103e365e14e3c13194ccbc2a5966f3/Documentation/man/accept.2.txt#L60-L69
